### PR TITLE
Support section footer actions

### PR DESCRIPTION
**Pull request by mischasigtermans**
## Description

This pull request adds a new footer section to the Section Blade component. This enhancement aims to improve the structure and flexibility of content presentation, allowing developers to include additional information, links, or actions at the bottom of each section.

screenshot-001252 (2024-02-17)@2x

## Code style

Add a slot named `footer` or `footerActions` to add the footer to a section. You may control the `footerActions` alignment with the `footerActionsAlignment` attribute on the Section component.

- [x] `composer cs` command has been run.

## Testing

Tested to work with `aside`, `collapsible` and `compact`.

- [x] Changes have been tested.

## Documentation

- [x] Documentation is up-to-date.

### CODE REVIEW

File: packages/support/resources/views/components/section/index.blade.php [hunk 1]
```diff
@@ -170,6 +184,49 @@
...
+                <div
+                    @if ($collapsible)
+                        x-bind:aria-expanded="(! isCollapsed).toString()"
+                    @if ($collapsed || $persistCollapsed)
+                        x-cloak
+                    @endif
+                    x-bind:class="{ 'invisible h-0 overflow-y-hidden border-none': isCollapsed }"
+                    @endif
+                    @class([
+                        'fi-modal-footer mt-6 border-t border-gray-200 dark:border-white/10',
```
Comment (zepfietje) [inline packages/support/resources/views/components/section/index.blade.php]: Please use correct `fi-*` CSS class naming related to this component please. 🙂

**Review by zepfietje — CHANGES_REQUESTED** [review]

**Comment by mischasigtermans** [comment]
@zepfietje I've discovered the behavior of the `headerActions` is not consistent in every case. I've changed the behavior and made sure works correctly in every case. Also made sure that when `collapsible` is set, the `headerActions` don't trigger collapsible.

screenshot-001268 (2024-02-20)@2x

**Comment by zepfietje** [comment]
@mischasigtermans, could you revert the header action changes and re-apply them in a separate PR?

There's been a couple times those actions broke because not all cases were accounted for, so I'd like to review it properly separately from the footer PR.

Thanks!

**Comment by mischasigtermans** [comment]
@zepfietje Done. Just tested all possibilities again for the footer. Let me know if I missed anything.

Header improvements PR: #11518

**Review by zepfietje — CHANGES_REQUESTED** [review]
If we're going to add footer actions support to the section Blade component, I think we need support for adding footer actions in the object API too. Could you add a `footerActions()` API similar to https://github.com/filamentphp/filament/pull/9161?

**Comment by mischasigtermans** [comment]
Not something I can do. Maybe @danharrin can squeeze it in?

**Comment by zepfietje** [comment]
What is unclear to you, @mischasigtermans?
Should be very similar to https://github.com/filamentphp/filament/pull/9161 if I'm correct. 🙂

**Comment by mischasigtermans** [comment]
It's a time management thing. 😉 Might be able to look at it in two weeks or so. Can it be a separate PR?

**Comment by zepfietje** [comment]
I understand, though I'm not sure if we should merge this in already without forms and infolists API support.

You're lucky that I just found the need for this myself. 😜
Will be taking over this PR, reviewing it and adding support for the new APIs. 👍

**Comment by zepfietje** [comment]
Replaced by https://github.com/filamentphp/filament/pull/11662, since I had to revert too many changes of the original PR. 🙂

**Pull request closed without merging**